operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/DependentResourceController.java

```diff
@@ -8,11 +8,8 @@
 import io.javaoperatorsdk.operator.processing.event.source.EventSource;
 
 public class DependentResourceController<R, P extends HasMetadata, C extends DependentResourceConfiguration<R, P>>
-    implements DependentResource<R, P>, Builder<R, P>, Updater<R, P>, Persister<R, P>,
-    Cleaner<R, P> {
+    implements DependentResource<R, P>, Persister<R, P>, Cleaner<R, P> {
 
-  private final Builder<R, P> builder;
-  private final Updater<R, P> updater;
   private final Cleaner<R, P> cleaner;
   private final Persister<R, P> persister;
```

Comment:
do we need the persister?

Reply:
Yes for non-Kubernetes resources, see the MySQL sample

Reply:
Not sure if it make sense for kubernetes objects; maybe we should have this just implicit in the reconcile function. Or just have it for a subclass for non-kubernetes objects and get called from reconcile in that impl.

Reply:
We could probably remove the Persister interface but I think that would require people defining Kubernetes dependents to either extend a class that would provide the behavior as it exists right now or do more work to implement the methods that are currently in Persister.